src/backend/opengl/PersistentPipelineStateGL.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BACKEND_OPENGL_PERSISTENTPIPELINESTATE_H_
+#define BACKEND_OPENGL_PERSISTENTPIPELINESTATE_H_
+
+#include "common/DepthStencilState.h"
+
+#include <bitset>
+
+namespace backend {
+namespace opengl {
+
+    class PersistentPipelineState {
```

Comment:
Per offline discussion this could be simplified to only by having only the following methods:

```
void SetDefaultState();
void SetStencilOpAndMasks(frontOp, backOp, readMask, writeMask);
void SetStencilReference(ref);
```

That check if the values changed and apply the state immediately if needed.